File: UVa/Java/Palinagram.java
import java.io.*;
import java.util.*;
public class Palinagram{
    public static void main(String[] args)throws IOException{
        BufferedReader key=new BufferedReader(new InputStreamReader(System.in));
        String str;
        while(!(str=key.readLine()).equals("#")){
            int[] freq=new int[128];
            for(char c : str.toCharArray())
                freq[c]++;
            ArrayList<Character> arr=new ArrayList<>();
            for(int i = 0; i < freq.length; i++){
                if(freq[i]%2!=0){
                    arr.add((char)i);
                }
            }
            Collections.sort(arr);
            String ret="";
            for(char c : arr){
                ret += c;
            }
            System.out.println(ret.substring(0, ret.length()-1));
        }
    }
}

File: UVa/C++/Palinagram.cpp
#include <iostream>
#include <vector>
#include <algorithm>

int main(){
    std::string str;
    while(true){
        if (std::cin >> str && str!="#"){
            int freq[129]={0};
            for(int i =0; i < str.length(); i++){
                freq[str[i]]++;
            }
            std::vector<char> vec;
            for(int i = 0; i <= 128; i++){
                if (freq[i]%2!=0){
                    vec.push_back((char)i);
                }
            }
            std::sort(vec.begin(), vec.end());
            std::string ret="";
            for(int i = 0; i <vec.size()-1; i++){
                ret += vec[i];
            }
            std::cout << ret << std::endl;
        }
        else{
            break;
        }
    }
}